src/services/codefixes/inferFromUsage.ts

```diff
@@ -1016,7 +1021,11 @@ namespace ts.codefix {
                     return !sigs.length || !checker.isTypeAssignableTo(source, getFunctionFromCalls(propUsage.calls));
                 }
                 else {
-                    return !checker.isTypeAssignableTo(source, combineFromUsage(propUsage));
+                    const combinedPropUsage = combineFromUsage(propUsage);
+                    if (combinedPropUsage.flags & TypeFlags.Void) {

```

Comment:
I think the correct fix for Too Much Void is to make `combineTypes` convert it to any when appropriate. Maybe combineFromUsage if it needs context, but from the example it seems like only a small amount of context might be needed: whether the property is in a call expression (void ok) or not (void not ok).